File: java/src/test/java/io/binac/leetcode/PlusOneTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class PlusOneTest {
    private static final PlusOne.Solution1 SOLUTION1 = new PlusOne.Solution1();

    private final int[] DIGITS1 = {1, 2, 3};
    private final int[] EXPECTED1 = {1, 2, 4};

    private final int[] DIGITS2 = {4, 3, 2, 1};
    private final int[] EXPECTED2 = {4, 3, 2, 2};

    private final int[] DIGITS3 = {9};
    private final int[] EXPECTED3 = {1, 0};


    @Test
    void test1() {
        assertArrayEquals(EXPECTED1, SOLUTION1.plusOne(DIGITS1));
        assertArrayEquals(EXPECTED2, SOLUTION1.plusOne(DIGITS2));
        assertArrayEquals(EXPECTED3, SOLUTION1.plusOne(DIGITS3));
    }
}

File: cpp/tests/PlusOneTest.cpp
#include "PlusOne.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(PlusOne, Solution66_1) {
  auto S66_1 = Solution66_1();

  std::vector<int> Digits1{1, 2, 3};
  const std::vector<int> Expected1{1, 2, 4};
  EXPECT_EQ(Expected1, S66_1.plusOne(Digits1));

  std::vector<int> Digits2{4, 3, 2, 1};
  const std::vector<int> Expected2{4, 3, 2, 2};
  EXPECT_EQ(Expected2, S66_1.plusOne(Digits2));

  std::vector<int> Digits3{9};
  const std::vector<int> Expected3{1, 0};
  EXPECT_EQ(Expected3, S66_1.plusOne(Digits3));
}